test/test-panel.js

```diff
@@ -1335,6 +1334,28 @@ exports["test panel load doesn't show"] = function*(assert) {
   loader.unload();
 }
 
+exports["test Panel without contentURL and contentScriptWhen=start should show"] = function*(assert) {
+  let loader = Loader(module);
+
+  let panel = loader.require("sdk/panel").Panel({
+    contentScriptWhen: "start",
+    // No contentURL, the bug only shows up when contentURL is not explicitly set.
+  });
+
+  let shown = false;
+  panel.once("show", function() {
+    shown = true;
+  });
+
+  panel.show();
+
+  yield wait(1000);
```

Comment:
replace this with:

    yield new Promise(resolve => {
      panel.once("show", resolve);
      panel.show();
    })